DESCRIPTION

```diff
@@ -1,13 +1,14 @@
 Package: MALDIquant
-Version: 1.19
+Version: 1.19.50
 Date: 2018-11-26
 Title: Quantitative Analysis of Mass Spectrometry Data
 Authors@R: c(person("Sebastian", "Gibb", role=c("aut", "cre"),
         email="mail@sebastiangibb.de",
         comment=c(ORCID="0000-0001-7406-4443")), person("Korbinian",
         "Strimmer", role="ths"))
+biocViews:
```

Comment:
Thanks for your reply/review Sebastian.

I just tested this thing; it does not work, although I found a lot of people suggesting this trick! I see your point on removing the dependency on matter. As you showed in your `onDiskVec` [example](https://gist.github.com/sgibb/8c8ace205ae566680cd08022b49fc790), we just need to write and read one vector at a time. The `onDiskVec` should have `path`, `offset`, `size` and `length` slots to account for `imzML` datatype.

Reply:
OK, if that don't work I don't really want to "Depend" or "Import" a Bioconductor package if it is not really necessary. If it would be possible to move it to "Suggests" it would be fine (but I am not sure that this is really possible).

I add a new branch: https://github.com/sgibb/MALDIquant/tree/OnDiskVector

If you like you could use and play with the new `OnDiskVector` class. It should work for both non-imzML and imzML data (it supports single files and offsets):
```r
library("MALDIquant")

odv <- OnDiskVector(1:10)
odv@path
# [1] "/tmp/Rtmpaz3SQG/file13ce3a31c136"
odv[1:5]
# [1] 1 2 3 4 5

f <- system.file(
    file.path("exampledata", "tiny_continuous.ibd"),
    package="MALDIquantForeign"
)
odv2 <- OnDiskVector(path=f, offset=56, n=5)
odv2@path
# [1] "/home/sebastian/opt/R/lib/R/library/MALDIquantForeign/exampledata/tiny_continuous.ibd"
odv2[1:5]
# [1]  6  7  8  9 10
```

Implementation could be found in https://github.com/sgibb/MALDIquant/blob/OnDiskVector/R/OnDiskVector-class.R

(Currently I have no time to really finish this thing. There is still the problem with `odv2 <- odv` and a subsequently different change of both vectors on the same file.)

Reply:
I use a similar approach as we discussed for `MSnbase` in https://github.com/lgatto/MSnbase/issues/429. TLDR; we store a modification counter in the object and in an additional file (in `MSnbase` we use the same file but that won't work for existing idb files). The modification counter is incremented if `[<-` is called, e.g.:
```
odv <- OnDiskVector(1:3)
odv2 <- odv
odv2[] <- 4:6
odv[]
# Error in .isModified.OnDiskVector(x9 :
#   /tmp/RtmpKh7TLm/file54ec578fcbd3 was modified by a different object.
```

Reply:
Thanks for your reply Sebastian & sorry for the delayed reply. 

So the way to go now, is to replicate whatever I did with the matter package but using the onDiskVector Class instead on the newly opened branch, right?

Reply:
If you satisfied with the `OnDiskVector` class that would be the way to go.